extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthMechanism.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.vertx.http.runtime.security;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Provides a way to select {@link HttpAuthenticationMechanism} used for a REST endpoint authentication.
+ * This annotation can only be used when proactive authentication is disabled. Using the annotation with
+ * enabled proactive authentication will lead to build-time failure.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.TYPE, ElementType.METHOD })
+@Inherited
+public @interface HttpAuthMechanism {
```

Comment:
```suggestion
public @interface HttpAuthentication {
```
or
```suggestion
public @interface HttpAuthenticationMechanism {
```

Reply:
BTW I had to move it to different package because such a class already exists there.